pkg/runtime/tags.go

```diff
@@ -52,18 +61,34 @@ func GetDefaultTags(
 func expandTagValue(
 	value *string,
 	obj rtclient.Object,
+	md acktypes.ServiceControllerMetadata,
 ) *string {
 	if value == nil || obj == nil {
 		return nil
 	}
 	var expandedValue string = ""
 	switch *value {
-	case "%UTCNOW%":
-		expandedValue = time.Now().UTC().String()
-	case "%KUBERNETES_NAMESPACE%":
+	case "%CONTROLLER_VERSION%":
+		expandedValue = generateControllerVersion(md)
```

Comment:
Could we split this into two different variables? Perhaps `%CONTROLLER_SERVICE%` and `%CONTROLLER_VERSION%`? This way users can, with more granularity, choose their tag formats.

Reply:
how about i tackle this in a followup PR?

This PR can deal with refactor of `serviceController` and addition of a new tag field. Keeping existing expandValue method as is.

Next PR will deal with supporting more reserved keywords, and find + replace keywords in all resource tags, not just the ones that match the switch case.

Reply:
The change I suggested will not be backward compatible. If we delay that change but push out this functionality, there is a chance that one of our GA controllers will be released between now and then with breaking compatibility. I'd prefer that if we are going to start enabling this feature, that we start off with the format we want to support in the long-term.

Reply:
Yeah, by followup PR, i meant next PR(before releasing this feature).

And I am curious to know what will be the backward incompatible change. I did not get that part.

Reply:
Oh the backward incompatibility would be `%CONTROLLER_VERSION%` changing from `s3-v1.0.0` to `v1.0.0`. But I guess as long as we get both of these out before cutting a new runtime release then I'm fine with it.

Reply:
After supporting these different tag variables, I was planning on changing the default tag to `services.k8s.aws/controller-version=%CONTROLLER_SERVICE%-%CONTROLLER_VERSION%` and then the output tag value will stay the same.

Also I wanted to do inplace substring replacement instead complete string replacement in the next PR.